stanford-earth/stanford_earth_saml
# Earth 904 Go to requested URL after logged in user closes/reopens browser

**READY FOR REVIEW**

**Summary**
When a logged in user closes the browser without first logging out, then reopens the browser and requests a specific page in the address bar, they are taken to the home page instead. Reason: simplesamlphp_auth module detects there is no longer a valid SAML session and calls Drupal user_logout then redirects to either fixed URI set in "Redirect after logout" field on "local authentication" tab of simplesamlphp_auth config settings or to home page if that field is blank.

The fix: 
1. sets stanford_earth_saml module weight to -100 so its hook_user_logout fires before simplesamlphp_auth's.
2. if destination is not '/' or 'user/logout', set a cookie with the requested destination.
3. creates a route for path '/redirectfromsamllogout' which can be used in the "Redirect after logout" field in simplesamlphp_auth.
4. the controller for the route checks for the cookie containing the originally requested URL and redirects there, or to the home page if no cookie.
5. this  version also fixes potential problems with the auto403 login feature already existing but to be turned on for intranet.
6. works with branch EARTH-904 of stanford-earth/se3_blt which includes configuration settings for dev, stage and prod.

**Needed By:** 
End of sprint, but sooner if possible

**Urgency**
Not particularly urgent.

**Steps to Test**
1. close and reopen a browser if already logged in
2. open browser and go to https://se3-dev.stanford.edu/inside/faculty-handbook
3. you should go through SAML login, then get to page.
4. close and reopen browser
5. go to https://se3-dev.stanford.edu/about/contact-us
6. you should get your page instead of going to the home page.
